### PR TITLE
Created params object that can parse json when asked too.

### PR DESCRIPTION
**Pull request by elorest**
### Description of the Change

Added a `Amber::Router::Params`.

This gives us a `#json(key)` method as well as allowing keys to be either symbols or strings.

Params work fine but validations seem messed up. I played with them a bit but not much progress

### CODE REVIEW

File: src/amber/validations/params.cr [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-  record Error, param : String, value : Amber::Router::ParamsType, message : String
+  record Error, param : String, value : Amber::Router::Params, message : String
 
   # This struct holds the validation rules to be performed
   class BaseRule
-    getter predicate : (Amber::Router::ParamsType -> Bool)
+    getter predicate : (Amber::Router::Params -> Bool)
     getter field : String
-    getter value : Amber::Router::ParamsType
+    getter value = Amber::Router::Params.new
```
Comment (elorest) [inline src/amber/validations/params.cr]: Before I changed this I was getting an error about it not being set by initialize.

**Comment by elorest** [comment]
All of my Real Life tests pass.

**Comment by eliasjpr** [comment]
It wsan't that bad :)

**Comment by elorest** [comment]
Thanks.

**Pull request merged**